v2/elasticsearch-common/src/main/java/com/google/cloud/teleport/v2/elasticsearch/transforms/ProcessFailsafePubSubFn.java

```diff
@@ -0,0 +1,64 @@
+package com.google.cloud.teleport.v2.elasticsearch.transforms;
+
+import com.google.cloud.teleport.v2.values.FailsafeElement;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.metrics.Counter;
+import org.apache.beam.sdk.metrics.Metrics;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.vendor.guava.v20_0.com.google.common.base.Throwables;
+
+/**
+ * The {@link ProcessFailsafePubSubFn} class processes a {@link FailsafeElement} containing an {@link PubsubMessage}
+ * and a String of the message's payload {@link PubsubMessage#getPayload()} into a {@link FailsafeElement} of
+ * the original {@link PubsubMessage} and a JSON string that has been processed with {@link Gson}.
+ *
+ * If {@link PubsubMessage#getAttributeMap()} is not empty then the message attributes will be serialized along with
+ * the message payload.
+ *
+ */
+public class ProcessFailsafePubSubFn
```

Comment:
Any idea why this class exists in the current module as it is unrelated to ElasticSearch transforms

Reply:
This class has been removed from elasticsearch-common.